system_metrics_collector/test/system_metrics_collector/test_linux_cpu_measurement.cpp

```diff
@@ -54,16 +102,100 @@ class TestLinuxCpuMeasurementNode : public system_metrics_collector::LinuxCpuMea
 private:
   system_metrics_collector::ProcCpuData makeSingleMeasurement() override
   {
-    first = !first;
+    EXPECT_GT(proc_samples.size(), measurement_index);
+    return processStatCpuLine(proc_samples[measurement_index++]);
+  }
 
-    if (first) {
-      return system_metrics_collector::processStatCpuLine(proc_sample_1);
-    } else {
-      return system_metrics_collector::processStatCpuLine(proc_sample_2);
+  int measurement_index{0};
+};
+
+class TestReceiveCpuMeasurementNode : public rclcpp::Node
+{
+public:
+  explicit TestReceiveCpuMeasurementNode(const std::string & name)
+  : rclcpp::Node(name), times_received(0)
+  {
+    auto callback = [this](MetricsMessage::UniquePtr msg) {this->MetricsMessageCallback(*msg);};
+    subscription = create_subscription<MetricsMessage,
+        std::function<void(MetricsMessage::UniquePtr)>>(TEST_TOPIC, 10, callback);
```

Comment:
```suggestion
        std::function<void(MetricsMessage::UniquePtr)>>(TEST_TOPIC, 10 /* something here */, callback);
```

Reply:
Done by adding comment that it means `history_depth`.